### PR TITLE
LoRA: Remove `ORTTrainer` support, embedding resize, `eval_dataset_size`, duplicate examples

### PR DESCRIPTION
**Pull request by jambayk**
## Describe your changes
**Data:**
- Rename `label_cols` to `label_col` and make its type string. We only use `label_cols[0]` which makes the name and list typing confusing

**LoRA/QLoRA:**
- Remove `smart_tokenizer_and_embedding_resize` which was used to add a pad_token if not present. This was copied from the original qlora implementation, but it is unnecessary and not efficient. Most training scripts use the eos_token as pad_token. This is fine as long as the attention mask and labels are correctly masked.
- Remove `eval_dataset_size` option which was used to split train data into train and eval. this is a specific use case which is supported using data configs that use slices of the train data. We want to remove as much of the data processing logic from the pass to make it generic
- Data collator is kept for now since the trainer has its own batch size (which depends on batch size params as well as number of gpus). It is more generic now.
- Remove support for `ORTTrainer`. This feature is not used and hasn't been validated for newer models or transformers version.
- Remove `phi` and `open_llama` qlora examples since these are copies of the examples for `llama`, `phi2` and `phi3`. Keep multiple copies of essentially the same workflow is hard to maintain when we make changes. 
- Other lora examples updated to remove redundant options.

## Checklist before requesting a review
- [ ] Add unit tests for this change.
- [ ] Make sure all tests can pass.
- [ ] Update documents if necessary.
- [ ] Lint and apply fixes to your code by running `lintrunner -a`
- [ ] Is this a user-facing change? If yes, give a description of this change to be included in the release notes.
- [ ] Is this PR including examples changes? If yes, please remember to update [example documentation](https://github.com/microsoft/Olive/blob/main/docs/source/examples.md) in a follow-up PR.

### CODE REVIEW

**Comment by jambayk** [comment]
Will split into multiple PRs
